### DISCUSSION TITLE
Issue with rewrites regex

### DISCUSSION THREAD

**Discussion by lriggle-strib** [vercel/next.js#70303] [Help]
### Summary

I'm trying to build a rewrite that accepts a URL that looks like this: `/some-path/12345`, and route it to a specific folder. I also have a rewrite that picks up things like `/some-path/some-deeper-path/the-deepest-path` and `/some-path/some-deeper-path`

```
[
  {
    source: '/:slug/:id(\\d{1,})', //for /some-path/12345
    destination: '/test/:slug/:id',
  },
  {
    source: '/:section*', //for everything else
    destination: '/test/:section*',
  },
]
```

Unfortunately I must be doing the regex wrong in my first rewrite because no matter what I do to try and indicate the `:id` parameter is numeric, `/some-path/some-deeper-path` still matches on it.

The regex pattern being used matches the example given on [this page](https://nextjs.org/docs/pages/api-reference/next-config-js/rewrites#regex-path-matching) in the Nextjs documentation. I must be missing something, however, because I cannot get the it to only match on numbers.

I have also tried using other regex patterns to indicate "numbers only, please", like `\d+`, `[0-9]+`, etc.

I've also looked through the somewhat sparse documentation on the `path-to-regexp` package and not found anything that works.

### Correct: /some-path
<img width="502" alt="Screenshot 2024-09-20 at 5 50 40 PM" src="https://github.com/user-attachments/assets/02dd0b8b-7144-44f7-b5af-fa5e27a05fb1">

### Incorrect: /some-path/some-deeper-path
<img width="562" alt="Screenshot 2024-09-20 at 5 50 34 PM" src="https://github.com/user-attachments/assets/13d52084-9685-4a55-a1da-0a3d8db9fed7">

### Correct: /some-path/some-deeper-path/the-deepest-path
<img width="697" alt="Screenshot 2024-09-20 at 5 50 48 PM" src="https://github.com/user-attachments/assets/beb078af-9073-4a6e-8544-455b9f4e427e">

### Correct: /some-path/12345
<img width="521" alt="Screenshot 2024-09-20 at 5 50 26 PM" src="https://github.com/user-attachments/assets/c3203761-71f9-44c6-92e9-ad1d0c8fc02f">

**Comment by icyJoseph** [comment]
I tried, directly with `path-to-regexp` using:

```js
const { pathToRegexp } = require("path-to-regexp");

const onlyDigits = "/:slug/:id([0-9]{1,})";

const paths = [
  "/some-path",
  "/some-path/some-deeper-path/the-deepest-path",
  "/some-path/12345",
  "/some-path/some-deeper-path",
];

const matcher = pathToRegexp(onlyDigits);

paths.forEach((path) => {
  console.log({ path, test: matcher.test(path) });
});

```

And I got:

```console
{ path: '/some-path', test: false }
{ path: '/some-path/some-deeper-path/the-deepest-path', test: false }
{ path: '/some-path/12345', test: true }
{ path: '/some-path/some-deeper-path', test: false }
```

Which I think, is what you want?
